build-tools/packages/build-cli/src/commands/merge/pullRequest.ts

```diff
@@ -0,0 +1,232 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Flags } from "@oclif/core";
+import { BaseCommand } from "../../base";
+import {
+	Repository,
+	createComment,
+	getPullRequestInfo,
+	listCommitsPullRequest,
+	mergePullRequest,
+} from "../../lib";
+import { CommandLogger } from "../../logging";
+
+interface CommitData {
+	sha: string;
+	commit: {
+		message: string;
+	};
+}
+
+interface PullRequestWithComment {
+	token: string;
+	owner: string;
+	repo: string;
+	prNumber: number;
+	comment: string;
+}
+
+interface PullRequestWithDetails {
+	title: string;
+	description: string;
+	mergeStrategy: MergeStrategy;
+}
```

Comment:
Based on the usage it may be more appropriate to call the first interface `IPullRequestDetails` and the second `IPullRequestMergeStrategy`. 

Also, maybe we consider having `IPullRequestDetails` have optional members `title`, `description`, and `mergeStrategy`. That way you can reuse `IPullRequestDetails` later (see below comment). There may be a better way to organize it though